### PR TITLE
Warn when non exactly parsing non floating-point

### LINKED ISSUE #11611: Parsing and displaying floating-point numbers up to approximation
I wonder whether this is moral or not that the following proof fails:
```
Require Import Floats.
Open Scope float_scope.
Goal 0.1 + 0.2 = 0.3.
Fail reflexivity.
```
and, conversely, that the following succeeds:
```
Require Import Floats.
Open Scope float_scope.
Goal 0.3 = 0.29999999999999997.
reflexivity.
Qed.
```

The documentation for [floats](https://coq.inria.fr/refman/language/coq-library.html?highlight=float#floats-library) says that `parse` of `print` is the identity thanks to the printer ensuring at least 17 digits and thus determining a unique float from its decimal approximation. Nothing is said about `print` of `parse` though.

The initial proposal about floats at #8276 was otherwise raising the question:

> Pretty-printing and parsing of floating-point values raises interesting questions. An easy solution would be to parse/print floats in hexadecimal format (e.g., `0xcp-3`) but this is ugly and perfectly unreadable for humans. A decimal format (e.g., `1.5`) would be more readable.
> It is known that using 17 digits while printing binary64 guarantees `parse \circ print : float -> float` to be the identity although `parse` is of course not injective.

Indeed, when reasoning with floating-point numbers, I guess that we quickly get familiar with having all numbers in the range [0.2999999999999997;0.30000000000000001] being bound to the same canonical representation 0.299999999999999988897769753748434595763683319091796875, so, the current approach is probably a good compromise for experts wanting to reason about floating-point numbers.

On the other side, I wonder if something more could be done for the non-specialist users who may be surprised by the discrepancy between what they type and what they see, and what is actually hidden behind the scene?

For instance, would it make sense that by default the parser warns about what exactly happens every time a float is parsed, indicating the full 54-digit denotation it shall be using internally and telling that for conciseness the printer will non-ambiguously write only the canonical first 17 digits of this denotation? Then an option would allow to deactivate this warning.

As a side remark, would it be better or on the contrary more confusing that `0.3` is displayed `0.3` rather than `0.2999999999999997`, since, after all, any member of the range is in principle good enough as a representative of the range and, if the purpose is conciseness, the printer could choose the shorter decimal representation?

Cc @erikmd, @proux01

### PR DESCRIPTION
**Pull request by proux01**
Following a suggestion from @herbelin in #11611 we emit a warning when parsing a value that is not exactly a floating-point value. No warning is emitted when the parsed value is exactly a floating-point value.

```
Coq < Require Import Floats.
Coq < Open Scope float_scope.
Coq < Check 1.25.
1.25
     : float
Coq < Check 0.3.
Toplevel input, characters 6-9:
> Check 0.3.
>       ^^^
Warning: The constant 0.3 is not a binary64 floating-point value. A closest
value will be used and unambiguously printed 0.29999999999999999.
[inexact-float,parsing]
```

~Depends on : #11703~

Fixes #11611 

If #8743 goes first, i'd be happy to rebase over it.

- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by Zimmi48** [comment]
Cool!

Note that it is usually a good idea when opening a PR that depends on another one to open this PR as "draft" which avoids triggering any review request.

**Comment by ppedrot** [comment]
@proux01 Shouldn't even that be an outright error? While it's kind of innocuous in standard programming languages, this looks like begging the Pollack inconsistency's stick to get beaten in Coq...

**Comment by JasonGross** [comment]
My inclination would be to have it be an error by default that can be turned into a warning.

**Comment by Zimmi48** [comment]
> My inclination would be to have it be an error by default that can be turned into a warning.

That's called a fatal-by-default warning.

**Comment by proux01** [comment]
Done so, following the popular demand.

I'm still not 100% sure this is the best thing to do though, as it means that some output can not be reparsed (for instance `Eval compute in (1 / 10)%float` will output `0.10000000000000001` which becomes unparsable).

> Note that it is usually a good idea when opening a PR that depends on another one to open this PR as "draft" which avoids triggering any review request.

Thanks for the advice!

**Comment by proux01** [comment]
Not really, the user can inspect floats functions such as `frexp` and `normfr_mantissa` and craft them precisely using `of_int63` and `ldexp`. But that's not a reasonable solution for everyday printing/parsing.

An intermediate solution would be to develop the hexadecimal notation `0x<hex_mantissa>p<binary_exp>` which would solve the parsing point. For printing, it could be used for instance with `Set Printing All`. Any other idea?

**Comment by ppedrot** [comment]
@proux01 I like the idea that `Set Printing All` outputs such a low-level representation. We could do that via a specialized flag part of the `All` set.

**Comment by Zimmi48** [comment]
> Not really, the user can inspect floats functions such as `frexp` and `normfr_mantissa` and craft them precisely using `of_int63` and `ldexp`. But that's not a reasonable solution for everyday printing/parsing.

That answers one recent question by @herbelin in #11646.

**Comment by erikmd** [comment]
@ppedrot 
> @proux01 Shouldn't even that be an outright error?

I don't this so: rounding is pervasive in floating-point arithmetic, including the decimal litterals themselves, but IMHO a warning would be enough (and a very nice feature anyway!)

In other words, I don't see why unlike other programming environments, we shouldn't be allowed to write `0.1%float` or `0.3%float` in Coq (e.g. if we port some C code directly in Coq and we want to prove it :). So, one such error can be distracting in practice especially if we have to restart Coq and pass a CLI flag to disable it! at least, a vernacular command that enable the error by default could be added to make this more convenient (?)

Regarding the Pollack inconsistency you mention, I should re-read Randy's paper to remember the various flavors of inconsistency he had defined, but AFAIK the situation here is not a critical one, given that formally, the litterals `0.29999999999999999` and `0.3` in scope `float` represent the same number (after the implied rounding). It can thus be viewed as an equivalence class, just like `5 mod 2` and `1 mod 2` represent the same number. And overall, there is no ambiguity w.r.t. the implied rounding as we only have binary64 (53-bit precision) floats in Coq.

To conclude due to the peculiarity of the decimal literals rounding, this feature is a nice step further for a formal floating-point arithmetic programming environment :+1:, but I'd suggest to:

1. either issue a warning only,
2. or an error by default that can be disabled by a `Set/Unset ...` vernacular (if this is easy to implement and use…)

@silene do you have a different opinion?

File: plugins/syntax/float_syntax.ml [hunk 1]
```diff
@@ -20,8 +20,58 @@ let make_path dir id = Libnames.make_path (make_dir dir) (Id.of_string id)
...
 
+let warn_inexact_float =
+  CWarnings.create ~name:"inexact-float" ~category:"parsing"
+    ~default:CWarnings.AsError
+    (fun (sn, f) ->
+      Pp.strbrk
+        (Printf.sprintf
+           "The constant %s is not a binary64 floating-point value. \
+            To turn this as a warning and use a closest value, which will \
+            unambiguously be printed %s, use option \"-w inexact-float\"."
```
Comment (SkySkimmer) [inline plugins/syntax/float_syntax.ml]: I think this should say something like "disable warning inexact-float" or otherwise avoid implying that going through the command line is necessary.

**Comment by ppedrot** [comment]
@erikmd going back and forth between error and warning modes can be done via a vernacular command, no need to go through the command-line! I am just advocating for the default to be an error.

**Comment by ppedrot** [comment]
> It can thus be viewed as an equivalence class

After having spent hours debugging horrible bugs in primitive projections, I know for a thing that there is nothing more frustrating than having mismatches between the internal representation and what is printed without any way to make them disappear... For debug, I think that a faithful representation is critical.

Also, I am slightly concerned that using Coq floats requires being an expert in the IEEE 754 specification. If my printer pretends that something is e.g. `1.000000001` but it's actually not because the output is not representable as a floating point number, I do hope that the computer is going to help me realizing that.

**Comment by erikmd** [comment]
> @erikmd going back and forth between error and warning modes can be done via a vernacular command, no need to go through the command-line! I am just advocating for the default to be an error.

I see; but in this case it'd be convenient that the "magic" Set vernacular to disable the error is mentioned in the error message… in addition to the message that would display the rounded floating-point number, for example.

File: plugins/syntax/float_syntax.ml (hunk 1 above)
Comment (proux01) [reply plugins/syntax/float_syntax.ml]: I agree. Would suggesting the vernacular `Set Warnings "inexact-float"` work? Something like "The constant %s is not a binary64 floating-point value. One can Set Warnings "inexact-float" to turn this into a warning and use a closest value, which will unambiguously be printed %s."

**Comment by erikmd** [comment]
@ppedrot 
> For debug, I think that a faithful representation is critical.
Agree.
To this aim, I think @proux01 's suggestions https://github.com/coq/coq/pull/11859#issuecomment-601235223 could be the way to go.

**Comment by erikmd** [comment]
However, I am concerned by this issue with the fatal-by-default warning raised by Pierre as well:

`Eval compute in (1 / 10)%float` will output `0.10000000000000001` which becomes unparsable.
so the Pollack inconsistency lies here actually.

So to sum up, maybe a mere non-fatal warning would be just OK…

**Comment by Zimmi48** [comment]
> I see; but in this case it'd be convenient that the "magic" Set vernacular to disable the error is mentioned in the error message… in addition to the message that would display the rounded floating-point number, for example.

This is more general than this specific warning, and a general request for improving the link between user messages and the documentation.

**Comment by proux01** [comment]
@ppedrot , just to be clear, the current representation *is* (somewhat) faithful in the sense that `1.000000001` can only be parsed as the underlying floating-point value (although it is not mathematically equal to the decimal number 1.000000001.

That being said, I fully understand your concern and I'd also like to have a more explicit representation for instance with Set Printing All (and it looks we all agree here). Therefore I offer the following solution:
1. I close the current PR
2. I develop hexadecimal numerals in a separate PR
3. I use them as lowest level representation for floats in a subsequent PR
4. I reopen and complete current PR

Would you agree with this plan?

@erikmd I'd also prefer the warning by default but I'm afraid we'll have to accept a consensus here.

**Comment by JasonGross** [comment]
> Eval compute in (1 / 10)%float will output 0.10000000000000001 which becomes unparsable.
so the Pollack inconsistency lies here actually.

This seems rather concerning.  Can someone explain this to me?  It seems to be that even in inexact mode, every float should have a unique decimal which it prints as and which is also the decimal which parses to that float with no warning.  (Why) is this not the case?

**Comment by proux01** [comment]
> > I see; but in this case it'd be convenient that the "magic" Set vernacular to disable the error is mentioned in the error message… in addition to the message that would display the rounded floating-point number, for example.
> 
> This is more general than this specific warning, and a general request for improving the link between user messages and the documentation.

Indeed, having a kind of link to the documentation about (un)setting warnings in the message would be great. I don't know how feasible it would be in practice though? Particularly in terms of verbosity and maintainability.

**Comment by Zimmi48** [comment]
> @erikmd I'd also prefer the warning by default but I'm afraid we'll have to accept a consensus here.

I would say this is up to you: you are the authors of the feature after all.

I personally think just a warning is fine.

I quite dislike the fact that *this specific warning* out of all the warnings that exist should explain how to disable itself.

**Comment by proux01** [comment]
> I quite dislike the fact that _this specific warning_ out of all the warnings that exist should explain how to disable itself.

Perfectly agree. I'm just concerned that if it's an error by default, the user may not even think it can be easily disabled.

**Comment by Zimmi48** [comment]
> Indeed, having a kind of link to the documentation about (un)setting warnings in the message would be great. I don't know how feasible it would be in practice though? Particularly in terms of verbosity and maintainability.

This seems very doable but more of a UI feature and not one that should make every warning message longer.

**Comment by proux01** [comment]
> This seems rather concerning. Can someone explain this to me? It seems to be that even in inexact mode, every float should have a unique decimal which it prints as and which is also the decimal which parses to that float with no warning. (Why) is this not the case?

Your understanding is perfectly good, it (currently) is the case. It would just no longer be if parsing a decimal constant that is not exactly a floating-point value is made an error.

**Comment by herbelin** [comment]
>  It would just no longer be if parsing a decimal constant that is not exactly a floating-point value is made an error.

We already had a similar discussion I think but could we consider that a floating-point number is not a value but an interval and that the parser/printer denotes the interval with the unique member of the interval which has the same 17 digits than the middle of the interval? (Assuming I'm not disregarding another subtlety of floating-point number...)

**Comment by erikmd** [comment]
> @erikmd I'd also prefer the warning by default but I'm afraid we'll have to accept a consensus here.

Again, having a goal that can't be parsed back if we copy-and-paste it, is very annoying AFAICT.

Also, thanks for proposing to look into the hexadecimal numeral approach! but I'm afraid it won't be feasible out-of-the-box because of the OCaml interface (given that your PR https://github.com/ocaml/ocaml/pull/8651 requires OCaml 4.10) ?
so if the Set-Printing-All-hexadecimal approach requires too much contorsions, maybe we could just keep this PR as is with a non-fatal-warning…

**Comment by JasonGross** [comment]
> Your understanding is perfectly good, it (currently) is the case. It would just no longer be if parsing a decimal constant that is not exactly a floating-point value is made an error.

Ah, I think I see.  There are some decimal constants that are not exactly representable as floats, and some floats that are not exactly representable as decimal constants.  The printing/parsing function still round-trips.  In light of this, I propose that there be two warnings:
1. If a decimal constant is parsed to a float which prints as a different decimal constant, emit a warning that is an error by default
2. If a decimal constant is parsed as a float which does not exactly represent the value of the decimal constant, but is still the unique decimal constant that this float gets printed as, print a different warning which is only a warning by default.

I do agree, though, that with `Set Printing All`, floats should print in the hex/binary exact representation

**Comment by erikmd** [comment]
@JasonGross 

> 1. If a decimal constant is parsed to a float which prints as a different decimal constant, emit a warning that is an error by default

OK why not, this allows one to avoid the unparseable issue (but then we won't be able to do a mere `Check 0.1%float.`)

> If a decimal constant is parsed as a float which does not exactly represent the value of the decimal constant, but is still the unique decimal constant that this float gets printed as, print a different warning which is only a warning by default.

OK.

> I do agree, though, that with Set Printing All, floats should print in the hex/binary exact representation

OK I don't object to that additional solution at all, I was just saying that with the current OCaml versions supported by Coq, it might be trickier (and maybe less efficient) to implement this ; and in addition the hex representation is not really readable (although one might object that it's not the purpose of `Set Printing All` ;-)

**Comment by proux01** [comment]
> I was just saying that with the current OCaml versions supported by Coq, it might be trickier (and maybe less efficient) to implement this on the Coq side;

Don't worry about that, it's just a matter of a few lines of OCaml if I remember well. The real work is in updating Coq lexer/parser and so on. For instance, would one accept being able to ask, say, `Show 0xf3`?

> and in addition the hex representation is not really readable (although one might object that it's not the purpose of `Set Printing All` ;-)

I object it's not the purpose of `Set Printing All` ;-)

**Comment by JasonGross** [comment]
> but then we won't be able to do a mere Check 0.1%float.

Indeed, I think this is a feature.  Users who know what they are doing (and, for example, want to prove things about an existing floating point program) can disable the error.  The message should definitely tell them what constant to use instead, and if that's all they were using `Check` for, this error won't make things harder.

**Comment by silene** [comment]
I did not read in detail all the messages, but there seems to be a misconception somewhere.

1. Not all decimal numbers are representable as floating-point numbers, obviously (cardinality argument). So, a warning is fine. It can even be fatal by default.

2. All floating-point numbers (in radix 2) are representable as decimal numbers! It is just that they can get insanely long (about 750 digits, plus the scientific notation). So, I would suggest some kind of threshold, e.g., display the decimal number if under 30 digits, display the hexadecimal number if over 30 digits.

**Comment by ppedrot** [comment]
Sorry for having derailed this PR. I think it is making the situation much better than before (a warning is better than nothing!), so I actually support merging it. Any other enhancement, like addition of a binary representation, can be done afterwards.

**Comment by Zimmi48** [comment]
So I suggest going back to the non-fatal warning for now and removing the additional explanation on changing the behavior of the warning from the message. It would be good if this new warning could be documented too (use `.. warn::` in the appropriate section of the manual).

File: doc/changelog/03-notations/11859-warn-inexact-float.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- **Added:**
+  Print a warning when parsing non floating-point values.
```
Comment (Zimmi48) [inline doc/changelog/03-notations/11859-warn-inexact-float.rst]: Out of context, this summary might seem a bit too vague. What about:
```suggestion
  In primitive floats, print a warning when parsing a decimal value that is not exactly a floating point number.
```

**Review by Zimmi48 — APPROVED** [review]

File: doc/changelog/03-notations/11859-warn-inexact-float.rst (hunk 2 above)
Comment (proux01) [reply doc/changelog/03-notations/11859-warn-inexact-float.rst]: Indeed, thanks!

File: doc/changelog/03-notations/11859-warn-inexact-float.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+- **Added:**
+  In primitive floats, print a warning when parsing a decimal value
+  that is not exactly a floating point number.
```
Comment (erikmd) [inline doc/changelog/03-notations/11859-warn-inexact-float.rst]: small nitpick:
```suggestion
  that is not exactly a binary64 floating-point number.
```

File: plugins/syntax/float_syntax.ml [hunk 4]
```diff
@@ -22,8 +22,56 @@ let make_path dir id = Libnames.make_path (make_dir dir) (Id.of_string id)
...
 (*** Parsing for float in digital notation ***)
 
+let warn_inexact_float =
+  CWarnings.create ~name:"inexact-float" ~category:"parsing"
+    (fun (sn, f) ->
+      Pp.strbrk
+        (Printf.sprintf
+           "The constant %s is not a binary64 floating-point value. \
+            A closest value will be used and unambiguously printed %s."
+           sn (Float64.to_string f)))
```
Comment (erikmd) [inline plugins/syntax/float_syntax.ml]: Apparently the message should be tuned to cope with the way infinities are really displayed in Coq:

```coq
Require Import Floats. Open Scope float_scope.
Check 1e1000.

> Check 1e1000.
>       ^^^^^^
Warning: The constant 1e1000 is not a binary64 floating-point value. A
closest value will be used and unambiguously printed inf.
[inexact-float,parsing]
infinity
     : float
```

and

```coq
Check -1e1000.

> Check -1e1000.
>       ^^^^^^^
Warning: The constant -1e1000 is not a binary64 floating-point value. A
closest value will be used and unambiguously printed -inf.
[inexact-float,parsing]
neg_infinity
     : float
```

But NaNs (especially the need to mask their sign) is not a issue here, given that a decimal literal won't be a NaN!

**Review by erikmd — CHANGES_REQUESTED** [review]
Thanks @proux01 ! The `inexact` function looks very fine.

I just wrote two suggestions in this review.

File: plugins/syntax/float_syntax.ml (hunk 4 above)
Comment (proux01) [reply plugins/syntax/float_syntax.ml]: fixed, please check though

**Review by erikmd — APPROVED** [review]

File: plugins/syntax/float_syntax.ml (hunk 4 above)
Comment (erikmd) [reply plugins/syntax/float_syntax.ml]: LGTM, thanks :+1:

File: kernel/float64.ml [hunk 5]
```diff
@@ -21,12 +21,17 @@ let is_neg_infinity f = f = neg_infinity
...
-   [nan] as input. *)
+   [nan] or an infinity as input. *)
 let to_string_raw f = Printf.sprintf "%.17g" f
 
 (* OCaml gives a sign to nan values which should not be displayed as
    all NaNs are considered equal here *)
-let to_string f = if is_nan f then "nan" else to_string_raw f
+let to_string f =
+  if is_nan f then "nan"
+  else if is_infinity f then "infinity"
```
Comment (SkySkimmer) [inline kernel/float64.ml]: Comment should be updated for the infinity printing
Something like `ocaml prints infinity as "inf" but we want "infinity"`

Comment (erikmd) [reply kernel/float64.ml]: Good idea.

BTW, do you think it is necessary to add an output test such as

```coq
Check 1e309%float.
Check (-1e309)%float.
```

?

Comment (proux01) [reply kernel/float64.ml]: Good idea, done, thanks!

**Comment by proux01** [comment]
> > but then we won't be able to do a mere Check 0.1%float.
> 
> Indeed, I think this is a feature. Users who know what they are doing (and, for example, want to prove things about an existing floating point program) can disable the error. The message should definitely tell them what constant to use instead, and if that's all they were using `Check` for, this error won't make things harder.

Sorry, haven't answered this yet. I first though this was a neat solution to the "output not reparsable" issue but I then realized it will, rather arbitrarily, split non binary64 decimal constants in two classes (the one used when printing and the others) which I dislike as it may give a false feeling that the formers are somewhat "less inexact" than the latters.

**Comment by proux01** [comment]
> 2. All floating-point numbers (in radix 2) are representable as decimal numbers! It is just that they can get insanely long (about 750 digits, plus the scientific notation). So, I would suggest some kind of threshold, e.g., display the decimal number if under 30 digits, display the hexadecimal number if over 30 digits.

I'm quite worried that this would almost always result in printing hexadecimal. For instance, for a constant as benign as `0.1`, the closest binary64 is `0.1000000000000000055511151231257827021181583404541015625` (55 digits, of which 70% don't carry any useful information).

**Comment by proux01** [comment]
> We already had a similar discussion I think but could we consider that a floating-point number is not a value but an interval and that the parser/printer denotes (without a warning nor an error) the interval with the unique member of the interval which has the same first 17 digits (base 10) than the middle of the interval? (Assuming I'm not disregarding another subtlety of floating-point numbers...)

That's a nice point of view. I'm not sure I'd like to advertise it too much though as it might give a false feeling to be performing interval arithmetic.

**Comment by silene** [comment]
> I'm quite worried that this would almost always result in printing hexadecimal.

Yes and no. If the user was able to input a number without triggering the warning (and without typing lots and lots of digits), then they will get back their decimal input. One should not underestimate how often integers or 1/2 or 1/4 appear in code or mathematical statements.

But anyway, this was just a counterpoint to people suggesting that `Set Printing All` should always display hexadecimal numbers.

**Comment by herbelin** [comment]
> That's a nice point of view. I'm not sure I'd like to advertise it too much though as it might give a false feeling to be performing interval arithmetic.

But we could say the same of floating-point numbers themselves. 17 printed digits does not mean that after many operations, all 17 digits are granted.

**Comment by erikmd** [comment]
> > That's a nice point of view. I'm not sure I'd like to advertise it too much though as it might give a false feeling to be performing interval arithmetic.
> 
> But we could say the same of floating-point numbers themselves. 17 printed digits does not mean that after many operations, all 17 digits are granted.

Actually I believe you both agree: floating-point operations such as `+, -, *, /` lead to a numerical value which is (1) guaranteed by the IEEE-754 to be *the rounding of the exact value*, but (2) when composing such operations, the final floating-point result is not guaranteed in terms of accuracy (i.e., regarding the **number of significant digits**) as that latter claim would require some error analysis.

On the other hand, interval arithmetic (which performs each operation twice, with directed roundings towards +∞ resp. -∞) do provides a guarantee that the exact mathematical result is contained in the final interval result of the composite operation (so e.g., one can factor-out the common digits of the two bounds to know the number of significant digits)

To sum up, even if a mere floating-point result could be viewed as a denotation for an interval, its value itself does not result from the use of interval-arithmetic algorithms… so I agree with @proux01 that displaying a mere float as an interval could be misleading here

**Comment by herbelin** [comment]
> so I agree with @proux01 that displaying a mere float as an interval could be misleading here.

I was not talking about displaying an interval but a different representative of the interval defined by a floating-point value. But, I agree, the question is in the "misleading".

There seems to be a compromise to find between what specialists of floating-point numbers find normal/misleading and what non-specialists find normal/misleading, and I'm fine with the current compromise.

**Comment by proux01** [comment]
Rebased, CI green

**Comment by erikmd** [comment]
@herbelin would you agree to be the assignee?
also this PR lacks a milestone; maybe 8.11.1 could be OK?
Kind regards, Erik

**Comment by herbelin** [comment]
OK, I assign and will merge tomorrow waiting for possible extra comments. For the milestone, ok, putting 8.11.1. Let's see what @ppedrot says.

**Comment by proux01** [comment]
Thanks to everybody who took part in the discussion!

This raised interesting points. I'll offer a low level representation for floats in a forthcoming PR.

**Comment by ppedrot** [comment]
@proux01 if this is supposed to be backported, I am afraid it'll need a fresh PR over v8.11 since the API has changed wildly since then.

**Comment by proux01** [comment]
@ppedrot Sure, can I do it by the end of the week?

**Comment by ppedrot** [comment]
That means it'll be postponed to 8.11.2, but it's probably not that important.

**Comment by proux01** [comment]
Otherwise I can try to do it tonight.

But I agree, it's not like a critical bugfix.

**Comment by ppedrot** [comment]
> Otherwise I can try to do it tonight.

Nah, it's really not important.

**Comment by proux01** [comment]
Well, it was easier than I feared in fact, here it is : https://github.com/coq/coq/pull/11975
Do whatever you want wih this PR.
